swift-sdk/Internal/InternalIterableAPI.swift

```diff
@@ -213,8 +213,12 @@ final class InternalIterableAPI: NSObject, PushTrackerProtocol, AuthProvider {
             return
         }
         
-        if !isEitherUserIdOrEmailSet() && config.enableAnonTracking && localStorage.userIdAnnon == nil {
-            anonymousUserManager.trackAnonTokenRegistration(token: token.hexString())
+        if !isEitherUserIdOrEmailSet() && localStorage.userIdAnnon == nil {
+            if config.enableAnonTracking {
+                anonymousUserManager.trackAnonTokenRegistration(token: token.hexString())
+            }
+            onFailure?("Iterable SDK must be initialized with an API key and user email/userId before calling SDK methods", nil)
```

Comment:
this failure message is inaccurate since the message could be triggered by the anonymous user tracking being disabled and also a criteria not being matched

Reply:
Hi @evantk91 this message is accurate because if you notice if condition message is triggered when SDK is NOT initialized. When SDK gets initialised it assigns null to localStorage.userIdAnnon anyways. 

Please let me know if you still have any doubts.

Reply:
The SDK initialization check needs to be separated and should stand alone. Right now, if anonymous user trracking is disabled, the track calls will be made regardless of if SDK is initialized. We should have a separate check that returns if SDK is not initialized.

Reply:
Also, the `localStorage.userIdAnnon` check can be removed. We want to update flow so that setUserId is called when  a known user is created. I will make a ticket for this if necessary.

Reply:
Hi @evantk91 If you notice line #221 it returns from there so in case SDK is not initialized track calls won't be made. We need to make API calls if localStorage.userIdAnnon is NOT NULL.

Can you please create a ticket for the new flow with an explanation so we know if removing localStorage.userIdAnnon makes sense or NOT?

Reply:
I will take a look. Can you separate out the SDK initialization check with its own return statement? We dont want the requestHandler to be called if the SDK is not initialized and anonymous tracking is disabled.

Reply:
I will create a new ticket if necessary.